subsys/net/l2/wifi/wifi_nm.c

```diff
@@ -117,7 +117,7 @@ int wifi_nm_register_mgd_type_iface(struct wifi_nm_instance *nm,
 	for (int i = 0; i < CONFIG_WIFI_NM_MAX_MANAGED_INTERFACES; i++) {
 		if (!nm->mgd_ifaces[i].iface) {
 			nm->mgd_ifaces[i].iface = iface;
-			nm->mgd_ifaces[i].type = (1 << type);
+			nm->mgd_ifaces[i].type = type;
```

Comment:
@MaochenWang1 had converted this to use bitmap, but I wasn't quite clear the usage of bitmap here, as the API always takes a single type and overrides the type. I am fine with this approach, @MaochenWang1  WDYT?

Reply:
I think using bit value makes sense here, the default value 0 would map to station.
I would like see the bit setting to use the existing `BIT` macro, so this should be written as
`nm->mgd_ifaces[i].type = BIT(type);`

Reply:
But if this bit value is causing issues somewhere else, then using plain value is probably ok too. It is not clear from the PR description what is the actual problem.

Reply:
It solves the problem in the following functions:
```
bool wifi_nm_iface_is_sta(struct net_if *iface)
{
    return wifi_nm_get_type_iface(iface) & WIFI_TYPE_STA;
}

bool wifi_nm_iface_is_sap(struct net_if *iface)
{
    return wifi_nm_get_type_iface(iface) & WIFI_TYPE_SAP;
}
```

The output of the above functions was something like this before these changes:

`(1 << type) & WIFI_TYPE_STA` and `(1 << type) & WIFI_TYPE_SAP`

where WIFI_TYPE_STA = 0 and WIFI_TYPE_SAP = 1. As the above functions are called in a loop, checking all interfaces until the required one is found, the actual behavior is as follows:

In the case of STA:
```
(1 << 0) & WIFI_TYPE_STA = 0, return 0
(1 << 1) & WIFI_TYPE_STA = 0, return 0
```
and the condition becomes false, resulted in returning default iface.

In the case of SAP:
```
(1 << 0) & WIFI_TYPE_SAP = 1, return 1
```
and the condition becomes true, where it was checking against `WIFI_TYPE_STA`. So, this was returning STA interface. which is a wrong interface. 

I hope you understand the problem that this PR addresses. I attempted to explain it using a dry run. Please let me know if my explanation was unclear, and I’ll be happy to clarify it further.

Reply:
This problem was fixed in PR: https://github.com/zephyrproject-rtos/zephyr/pull/75982.

Reply:
(1 << xyz) to use BIT(xyz)

Reply:
Since your original issue is already fixed, can you update your commit message about your change of 

> updated (1 << xyz) to use BIT(xyz)?